src/herder/TransactionQueue.h

```diff
@@ -61,88 +47,53 @@ class TransactionQueue
-     */
+#ifdef BUILD_TESTS
     struct AccountTxQueueInfo
     {
         SequenceNumber mMaxSeq{0};
         int64_t mTotalFees{0};
         int mAge{0};
-
-        friend bool operator==(AccountTxQueueInfo const& x,
-                               AccountTxQueueInfo const& y);
     };
+    AccountTxQueueInfo
+    getAccountTransactionQueueInfo(AccountID const& acc) const;
+#endif
 
-    /**
-     * Queue of transaction for given account. mTotalFees is a sum of all
-     * feeBid() values from mTransactions. mAge is incremented each time
-     * shift() is called and allows for banning transactions.
-     */
+  private:
     struct AccountTransactions
     {
-        using Transactions = std::vector<TransactionFrameBasePtr>;
+        using Transactions = std::vector<std::vector<TransactionFrameBasePtr>>;
 
-        int64_t mTotalFees{0};
         int mAge{0};
         Transactions mTransactions;
     };
 
-    explicit TransactionQueue(Application& app, int pendingDepth, int banDepth);
-
-    AddResult tryAdd(TransactionFrameBasePtr tx);
-    void removeAndReset(std::vector<TransactionFrameBasePtr> const& txs);
-    void ban(std::vector<TransactionFrameBasePtr> const& txs);
+    Application& mApp;
+    size_t mPendingDepth;
+    std::vector<medida::Counter*> mSizeByAge;
 
-    /**
-     * Increse age of each transaction queue. If that age now is equal to
-     * pendingDepth, all ot transaction on that queue are banned. Also
-     * increments age for each banned transaction and if that age became equal
-     * to banDepth, transaction get unbanned.
-     */
-    void shift();
+    using PendingTransactions =
+        std::unordered_map<AccountID, AccountTransactions>;
+    PendingTransactions mPendingTransactions;
 
-    AccountTxQueueInfo
-    getAccountTransactionQueueInfo(AccountID const& accountID) const;
+    std::unordered_map<AccountID, int64_t> mTotalFees;
 
-    int countBanned(int index) const;
-    bool isBanned(Hash const& hash) const;
+    using BannedTransactions = std::deque<std::unordered_set<Hash>>;
+    BannedTransactions mBannedTransactions;
 
-    std::shared_ptr<TxSetFrame> toTxSet(Hash const& lclHash) const;
+    void addFee(TransactionFrameBasePtr const& tx);
 
-  private:
-    /**
-     * Per account queue. Each queue has its own age, so it is easy to reset it
-     * when transaction for given account was included in ledger. It also
-     * allows for fast banning of all transaction that depend (have bigger
-     * sequence number) of just-removed invalid one in ban().
-     */
-    using PendingTransactions =
-        std::unordered_map<AccountID, AccountTransactions>;
-    /**
-     * Banned transactions are stored in deque of depth banDepth, so it is easy
-     * to unban all transactions that were banned for long enoug.
-     */
-    using BannedTransactions = std::deque<std::unordered_set<Hash>>;
+    AccountTransactions::Transactions::iterator
+    findTx(AccountTransactions::Transactions& transactions,
```

Comment:
`findTx` is not a great name. It took me a while to understand that what it's looking for is not a transaction but the `slot` in the transaction queue that corresponds to that tx